samples/pmic/npmx/buck/README.rst

```diff
@@ -0,0 +1,139 @@
+.. _npmx_buck_sample:
+
+npmx: BUCK
+##########
+
+.. contents::
+   :local:
+   :depth: 2
+
+The npmx BUCK sample demonstrates how to control PMIC BUCKs using npmx drivers.
+
+Requirements
+************
+
+The sample supports the following development kits:
+
+.. table-from-rows:: /includes/sample_board_rows.txt
+   :header: heading
+   :rows: nrf52840dk_nrf52840, nrf5340dk_nrf5340_cpuapp
+
+The sample also requires an nPM1300 EK.
+
+Overview
+********
+
+This sample uses npmx BUCK driver to:
+  * set BUCK output voltage
+  * set BUCK retention voltage with GPIO
+  * control BUCK enable mode with GPIO
+
+Wiring
+******
+
+Connect the TWI interface between the chosen DK and the nPM1300 EK as in the following table:
+
+.. list-table:: nPM1300 EK connections.
+   :widths: 25 25 25
+   :header-rows: 1
+
+   * - nPM1300 EK
+     - nRF5340 DK
+     - nRF52840 DK
+   * - GPIO0
+     - P1.10
+     - P1.10
+   * - SDA
+     - P1.02
+     - P0.26
+   * - SCL
+     - P1.03
+     - P0.27
+   * - VOUT2 & GND
+     - External supply (P21)
+     - External supply (P21)
+
+Make the following connections on the nPM1300 EK:
+
+* Connect a USB power supply to the **J3** connector.
+* Connect a suitable battery to the **J2** connector.
+* On the **P18** pin header, connect **VOUT2** and **VDDIO** pins with a jumper.
+* On the **P2** pin header, connect **VBAT** and **VBATIN** pins with a jumper.
+* On the **P17** pin header, connect all LEDs with jumpers.
+* On the **P13** pin header, connect **RSET1** and **GND** pins with a jumper.
+* On the **P14** pin header, connect **RSET2** and **VSET2** pins with a jumper.
+
+Make the following connections on the DK:
+
+* Switch nRF power source **SW9** to **VDD**.
+* Switch VEXT -> nRF **SW10** to **ON**.
+
+Configuration
+*************
+
+|config|
+
+Configuration options
+=====================
+
+Check and configure the following Kconfig options to specify which testcase to run:
+
+.. TESTCASE_SET_BUCK_VOLTAGE:
+
+TESTCASE_SET_BUCK_VOLTAGE
+  This sample configuration enables simple output voltage testcase.
+  Voltage is set to 2.4 V (`NPMX_BUCK_VOLTAGE_2V4`) on BUCK 1.
+
+.. TESTCASE_OUTPUT_VOLTAGE:
+
+TESTCASE_OUTPUT_VOLTAGE
+  This sample configuration enables advanced output voltage testcase.
+  BUCK 1 is starting from voltage 1.0 V (`NPMX_BUCK_VOLTAGE_1V0`).
+  The output voltage will increase from the selected to the maximum voltage (3.3 V).
+
+.. TESTCASE_RETENTION_VOLTAGE:
+
+TESTCASE_RETENTION_VOLTAGE
+  This sample configuration enables retention voltage testcase.
+  GPIO1 (`NPMX_GPIO_INSTANCE_1`) is set to input mode and configured to be used as a retention input selector for BUCK 1 (`NPMX_BUCK_INSTANCE_1`).
+  Normal mode voltage is set to 1.7 V (`NPMX_BUCK_VOLTAGE_1V7`) and retention voltage to 3.3 V (`NPMX_BUCK_VOLTAGE_3V3`).
+  When you change the input state of the selected GPIO you should see voltage changes in the BUCK 1 output.
+
+.. TESTCASE_ENABLE_BUCKS_VIA_PIN:
+
+TESTCASE_ENABLE_BUCKS_VIA_PIN
```

Comment:
Is it possible to change these names to avoid using **via** word? Can we change it to **through**?